### DISCUSSION TITLE
Recommendation. New frag grenade code.

### DISCUSSION THREAD

**Discussion by SydWad** [FakeFishGames/Barotrauma#15253] [Unstable feedback]
Currently the new frag grenade code when it comes to being triggered by detonators seems rather limited, and has some compatibility problems with older mods. I have a simple suggestion to fix that.
Currently it uses:`

      <StatusEffect type="OnUse" targettype="This" Condition="0" setvalue="true" comparison="OR">
        <Conditional HasTag="detonator" targetcontainer="true" />
        <Conditional HasTag="railgunammo" targetcontainer="true" />
        <Conditional HasTag="depthchargeammo" targetcontainer="true" />
      </StatusEffect>
      `
When it could just use:`

      <StatusEffect type="OnUse" targettype="This" Condition="0" setvalue="true">
        <Conditional entitytype="Item" targetcontainer="true" />
      </StatusEffect>
      `
Then any item that triggers it will cause it to blow instantly, similar to how it used to be.
As far as i can think of there isnt a single case where an item triggering an explosive wouldn't want to be blowing it up.

**Comment by SydWad** [comment]
hello? Been two years and i would still like this to be changed. Modded turret ammo doesnt work with frag grenades still.

**Reply by SoobGorl** [reply]
I'm not sure I understand what you mean. Arming Frag Grenades in a detonator (the item) causes them to explode (though with a delay, like all detonator-compatible items--changing this functionality to be instant is up to modders to override the vanilla xml assembly to better match with older behavior). What does modded turret ammo have to do with frag grenades, in this context (do you mean putting them inside things like railguns? Because shells do detonate if a frag grenade is placed into them.)?

**Reply by SydWad** [reply]
I didnt change anything about the activation time. What i changed is its a much cleaner solution for the conditionals checking what container the grenade is inside of.
The issue is if anything falls outside of this list the vanilla grenade wont recognize that and wont be triggered in the way its meant to.
